3.10/aql/operations-replace.md

```diff
@@ -25,104 +25,171 @@ REPLACE <em>keyExpression</em> WITH <em>document</em> IN <em>collection</em></co
 ```
 
-When using the second syntax, `keyExpression` provides the document identification.
-This can either be a string (which must then contain the document key) or a
-document, which must contain a `_key` attribute.
+### `REPLACE <keyExpression> WITH <document> IN <collection>`
 
-The following queries are equivalent:
+Using the second syntax, the document to replace is defined by the
+`keyExpression`. It can either be a string with the document key, an object
+which contains a `_key` attribute with the document key, or an expression that
+evaluates to either of these two.
+
+The following query replaces the document identified by the key `my_key` in the
+`users` collection, only setting a `name` attribute. The key is passed as
+a string in the `keyExpression`. The attributes to set are passed
+separately as the `document` object:
 
 ```aql
-FOR u IN users
-  REPLACE { _key: u._key, name: CONCAT(u.firstName, u.lastName) } IN users
+REPLACE "my_key" WITH { name: "Jon", status: "active" } IN users
 ```
 
+The `document` object may contain a `_key` attribute, but it is ignored.
+
+You cannot define the document to replace using an `_id` attribute, nor pass a
+document identifier as a string (like `"users/john"`). However, you can use
+`PARSE_IDENTIFIER(<id>).key` as `keyExpression` to get the document key as a
+string.
+
+### Comparison of the syntaxes
+
+The following queries are equivalent:
+
 ```aql
 FOR u IN users
-  REPLACE u._key WITH { name: CONCAT(u.firstName, u.lastName) } IN users
+  REPLACE u._key WITH { name: CONCAT(u.firstName, " ", u.lastName), status: u.status } IN users
 ```
 
 ```aql
 FOR u IN users
-  REPLACE { _key: u._key } WITH { name: CONCAT(u.firstName, u.lastName) } IN users
+  REPLACE { _key: u._key } WITH { name: CONCAT(u.firstName, " ", u.lastName), status: u.status } IN users
 ```
 
 ```aql
 FOR u IN users
-  REPLACE u WITH { name: CONCAT(u.firstName, u.lastName) } IN users
+  REPLACE u WITH { name: CONCAT(u.firstName, " ", u.lastName), status: u.status } IN users
 ```
 
-A replace will fully replace an existing document, but it will not modify the values
-of internal attributes (such as `_id`, `_key`, `_from` and `_to`). Replacing a document
-will modify a document's revision number with a server-generated value.
+Dynamic key expressions
+-----------------------
 
-A replace operation may update arbitrary documents:
+An `REPLACE` operation may replace arbitrary documents, using either of the two
+syntaxes:
+
+```aql
+FOR i IN 1..1000
+  REPLACE { _key: CONCAT('test', i), name: "Paula", status: "active" } IN users
+```
 
 ```aql
 FOR i IN 1..1000
-  REPLACE CONCAT('test', i) WITH { foobar: true } IN users
+  REPLACE CONCAT('test', i) WITH { name: "Paula", status: "active" } IN users
 ```
 
-The documents it modifies can be in a different collection than
-the ones produced by a preceding `FOR` operation:
+Target a different collection
+-----------------------------
+
```

Comment:
```suggestion
Recall the example:
```aql
FOR u IN users
  REPLACE u WITH { name: CONCAT(u.firstName, " ", u.lastName), status: u.status } IN users
```
Although the `u` variable holds a whole document, it is used only to define the `_key` attribute in the `REPLACE` operation. The latter then finds a document defined by the `_key` attribute in the collection defined in the `IN` clause and replaces it with the contents of the `REPLACE` clause. Of course, if `FOR` iterates over the same collection, the document held in the variable `u` will be found and replaced. However, `FOR` can also iterate over another collection:
```aql
FOR u in users
   REPLACE u WITH { name: CONCAT(u.firstName, " ", u.lastName), status: u.status } IN backup
```
Then the replaced document is different from `u` (it is in another collection) and only shares the `_key` attribute with it.
```